### PR TITLE
Set expected value of line-height to expanded generic event list summary (GELS) spacer on IRC layout

### PR DESCRIPTION
**Pull request by luixxiul**
Closes https://github.com/vector-im/element-web/issues/24605

This PR replaces the inherited value of `line-height` of expanded generic event list summary's spacer on IRC layout with the presumably expected one (`$irc-line-height` specified with `!important`). It also adds tests for spacers on IRC layout, compact modern/group layout, and bubble layout which does not display the spacer. Please note this PR does not change the clickable area of `expand` and `collapse` link buttons.

|Before|After|
|---------|------|
|Screenshot from 2023-02-22 04-07-25|Screenshot from 2023-02-22 04-09-10|
|Screenshot from 2023-02-22 04-06-18|Screenshot from 2023-02-22 04-03-49|

https://user-images.githubusercontent.com/3362943/220442868-f168feae-756c-422a-b9e1-4ac1e980b123.mp4

type: defect
Notes: Reduce height reserved for "collapse" button's line on IRC layout

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Reduce height reserved for "collapse" button's line on IRC layout ([\#10211](https://github.com/matrix-org/matrix-react-sdk/pull/10211)). Fixes vector-im/element-web#24605. Contributed by @luixxiul.

### CODE REVIEW

File: res/css/views/elements/_GenericEventListSummary.pcss [hunk 1]
```diff
@@ -21,6 +21,16 @@ limitations under the License.
         margin-right: $spacing-8;
     }
 
+    .mx_GenericEventListSummary_spacer {
+        border-bottom: 1px solid $primary-hairline-color;
+        margin-left: 63px; /* TODO: Use a varilable */
+        line-height: $font-30px; /* TODO: Use a varilable */
+
+        .mx_IRCLayout & {
+            line-height: var(--line-height);
```
Comment (luixxiul) [inline res/css/views/elements/_GenericEventListSummary.pcss:30]: This declaration has the expected line-height applied to the spacer line against `line-height: $font-30px;`.

File: cypress/e2e/timeline/timeline.spec.ts [hunk 2]
```diff
@@ -279,6 +349,14 @@ describe("Timeline", () => {
...
             cy.get(".mx_GenericEventListSummary_toggle[aria-expanded=false]").click();
 
+            // Make sure the "expand" link button worked
+            cy.get(".mx_GenericEventListSummary_toggle[aria-expanded=true]").should("exist");
+
+            // Make sure spacer is not visible on bubble layout
+            cy.get(".mx_GenericEventListSummary[data-layout=bubble] .mx_GenericEventListSummary_spacer").should(
+                "not.be.visible", // See: _GenericEventListSummary.pcss
+            );
+
```
Comment (luixxiul) [inline cypress/e2e/timeline/timeline.spec.ts:359]: I'm adding a percy test here

File: res/css/views/elements/_GenericEventListSummary.pcss [hunk 3]
```diff
@@ -21,6 +21,16 @@ limitations under the License.
         margin-right: $spacing-8;
     }
 
+    .mx_GenericEventListSummary_spacer {
+        border-bottom: 1px solid $primary-hairline-color;
+        margin-left: 63px; /* TODO: Use a varilable */
+        line-height: $font-30px; /* TODO: Use a varilable */
```
Comment (gsouquet) [inline res/css/views/elements/_GenericEventListSummary.pcss:27]:
```suggestion
        margin-left: 63px; /* TODO: Use a variable */
        line-height: $font-30px; /* TODO: Use a variable */
```

**Review by gsouquet — APPROVED** [review]
Looks good to me! Thank you for the update.
Could you update the typo in the comments? And we'll be able to merge this

File: res/css/views/elements/_GenericEventListSummary.pcss (hunk 3 above)
Comment (luixxiul) [reply res/css/views/elements/_GenericEventListSummary.pcss:27]: Thanks for finding them! I didn't notice at all..

File: cypress/e2e/timeline/timeline.spec.ts (hunk 2 above)
Comment (luixxiul) [reply cypress/e2e/timeline/timeline.spec.ts:361]: Added with https://github.com/matrix-org/matrix-react-sdk/pull/10211/commits/bc507c743809cf17d857bffae5525795661be5d0 along with another one.

**Comment by luixxiul** [comment]
@gsouquet Would you please re-review so that the test which this PR is going to add will be available for https://github.com/matrix-org/matrix-react-sdk/pull/10189? Thanks in advance.

**Review by justjanne — APPROVED** [review]

**Comment by luixxiul** [comment]
https://github.com/matrix-org/matrix-react-sdk/pull/10211#pullrequestreview-1332410346 Thanks for the review!

**Comment by luixxiul** [comment]
@andybalaam Please note that this PR introduced Percy test changes logged here (with a possibly upstream bug): https://github.com/vector-im/element-web/issues/24949. I think those changes are safe to approve.

**Pull request merged**